File: src/Controllers/Controllers/ProductsController.cs
using Microsoft.AspNetCore.Mvc;
using Controllers.Models;

namespace Controllers
{
	/// <summary>
	/// API controller that provides product information and update endpoints.
	/// </summary>
    [TypeScriptController]
    [Route("products")]
	public class ProductsController : Controller
    {
		/// <summary>
		/// Get data page from list of all products.
		/// </summary>
		///	<param name="page">Pagination page.</param>
		///	<param name="limit">Pagination limit per page.</param>
		///	<returns>A page of products.</returns>
		[HttpGet, Route("all/{limit}/{page}")]

        public JsonResult<Product[]> GetAllProducts(int page, int limit)
        {
			var result = new Product[]
				{
					new Product
					{
						Name = "Abba"
					},
					new Product
					{
						Name = "Cabba"
					},
				};

			return new JsonResult<Product[]>(result);
        }

		/// <summary>
		/// Update given product.
		/// </summary>
		///	<param name="product">New product data.</param>
		///	<returns>State of transaction.</returns>
		[HttpPost, Route("update")]
        public JsonResult<Product[]> UpdateProduct(Product product)
        {
			var result = new Product[]
				{
					new Product
					{
						Name = "Abba"
					},
					new Product
					{
						Name = "Cabba"                                                                                                                             ,
					},
				};

			return new JsonResult<Product[]>(result);
        }
    }
}

File: src/Controllers/scripts/controllers/ProductsController.ts
/*
 * TypeWriter generated API controller for Controllers.ProductsController
 */

/* Imported models */
import { Product } from '../models/Product';

/* Imported controller */
import { Controller } from './controller';

/* Generated class */
const controllerRoute = 'products';

/**
 * API controller that provides product information and update endpoints.
 * @extends Controller
 */
export class ProductsController extends Controller {

    /**
      * Get data page from list of all products.
      * @param page Pagination page.
      * @param limit Pagination limit per page.
      * @returns {Promise} A page of products.
      */
	public getAllProducts(
			page: number,
			limit: number) {
		const parameters = {
			'page': page,
			'limit': limit,
		};
		const actionRoute = 'all/{limit}/{page}';
		const route = controllerRoute + '/' + actionRoute;

		return this.fetchGet<Product[]>(route, parameters);
	}

    /**
      * Update given product.
      * @param product New product data.
      * @returns {Promise} State of transaction.
      */
	public updateProduct(
			product: Product) {
		const parameters = {
			'product': product,
		};
		const actionRoute = 'update';
		const route = controllerRoute + '/' + actionRoute;

		return this.fetchPost<Product[]>(route, parameters);
	}
}
